Faking a UUID panics

I had a complex struct generating a panic from faker and I isolated it to the uuid field causing the issue.
Here's the minimal code to show the error:

```
package main

import (
	"github.com/bxcodec/faker"
	"github.com/gofrs/uuid"
	"fmt"
)

func main() {
	var u uuid.UUID
	faker.FakeData(&u)
	fmt.Println(u)
}
```

And the error trace is:
```
panic: reflect.MakeSlice of non-slice type

goroutine 1 [running]:
reflect.MakeSlice(0x11367c0, 0x1104fe0, 0x58, 0x58, 0xc420065cc8, 0xc420065cd8, 0x1010068)
	/usr/local/Cellar/go/1.10.3/libexec/src/reflect/value.go:2092 +0x1ce
github.com/bxcodec/faker.getValue(0x11367c0, 0x1104fe0, 0x1104fe0, 0xc4200aa250, 0x16, 0x0, 0x10)
	/Users/msorens/code/go/src/github.com/bxcodec/faker/faker.go:218 +0x7a9
github.com/bxcodec/faker.FakeData(0x110c840, 0xc4200aa250, 0xc42008a058, 0x0)
	/Users/msorens/code/go/src/github.com/bxcodec/faker/faker.go:145 +0x1bc
main.main()
	/Users/msorens/code/go/src/github.com/me/fake/tryfake.go:11 +0x4b
exit status 2
```

--- a/.gitignore
+++ b/.gitignore
@@ -10,6 +10,8 @@
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
 cover.html
+*/.vscode
+
 
 # Project-local glide cache, RE: https://github.com/Masterminds/glide/issues/736
 .glide/

--- a/.gitignore
+++ b/.gitignore
@@ -10,7 +10,7 @@
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
 cover.html
-*/.vscode
+.vscode/
 
 
 # Project-local glide cache, RE: https://github.com/Masterminds/glide/issues/736

--- a/README.md
+++ b/README.md
@@ -93,6 +93,10 @@ Supported tag:
 * Amount
 * Amount with Currency
 
+**UUID :**
+* UUID Digit (32 bytes)
+* UUID Hyphenated (36 bytes)
+
 **Skip :**
 * \-
 
@@ -146,6 +150,8 @@ type SomeStruct struct {
 	Currency           string  `faker:"currency"`
 	Amount             float64 `faker:"amount"`
 	AmountWithCurrency string  `faker:"amount_with_currency"`
+	UUIDHypenated	   string  `faker:"uuid_hyphenated"`
+	UUID	           string  `faker:"uuid_digit"`
 	Skip		   string  `faker:"-"`
 }
 
@@ -198,6 +204,8 @@ func main() {
 			Currency: IRR,
 			Amount: 88.990000,
 			AmountWithCurrency: XBB 49257.100000,
+			UUIDHypenated: 8f8e4463-9560-4a38-9b0c-ef24481e4e27,
+			UUID: 90ea6479fd0e4940af741f0a87596b73,
 			Skip:
 		}
 	*/

--- a/faker.go
+++ b/faker.go
@@ -20,8 +20,8 @@ const (
 	letterIdxMax       = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
 	letterBytes        = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
 	tagName            = "faker"
-	UUIDDigit          = "uuid_digit"
-	UUIDHyphenated     = "uuid_hyphenated"
+	ID                 = "uuid_digit"
+	HyphenatedID       = "uuid_hyphenated"
 	Email              = "email"
 	MacAddress         = "mac_address"
 	DomainName         = "domain_name"
@@ -104,8 +104,8 @@ var mapperTag = map[string]interface{}{
 	Currency:           GetPrice().Currency,
 	Amount:             GetPrice().Amount,
 	AmountWithCurrency: GetPrice().AmountWithCurrency,
-	UUIDDigit:          GetIdentifier().Digit,
-	UUIDHyphenated:     GetIdentifier().Hyphenated,
+	ID:                 GetIdentifier().Digit,
+	HyphenatedID:       GetIdentifier().Hyphenated,
 }
 
 // Generic Error Messages for tags

--- a/faker.go
+++ b/faker.go
@@ -104,8 +104,8 @@ var mapperTag = map[string]interface{}{
 	Currency:           GetPrice().Currency,
 	Amount:             GetPrice().Amount,
 	AmountWithCurrency: GetPrice().AmountWithCurrency,
-	UUIDDigit:          GetUUID().Digit,
-	UUIDHyphenated:     GetUUID().Hyphenated,
+	UUIDDigit:          GetIdentifier().Digit,
+	UUIDHyphenated:     GetIdentifier().Hyphenated,
 }
 
 // Generic Error Messages for tags

--- a/faker.go
+++ b/faker.go
@@ -20,6 +20,8 @@ const (
 	letterIdxMax       = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
 	letterBytes        = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
 	tagName            = "faker"
+	UUIDDigit          = "uuid_digit"
+	UUIDHyphenated     = "uuid_hyphenated"
 	Email              = "email"
 	MacAddress         = "mac_address"
 	DomainName         = "domain_name"
@@ -102,6 +104,8 @@ var mapperTag = map[string]interface{}{
 	Currency:           GetPrice().Currency,
 	Amount:             GetPrice().Amount,
 	AmountWithCurrency: GetPrice().AmountWithCurrency,
+	UUIDDigit:          GetUUID().Digit,
+	UUIDHyphenated:     GetUUID().Hyphenated,
 }
 
 // Generic Error Messages for tags

--- a/faker_test.go
+++ b/faker_test.go
@@ -166,7 +166,7 @@ func (t TaggedStruct) String() string {
 		t.DayOfMonth, t.Timestamp, t.Century, t.TimeZone,
 		t.TimePeriod, t.Word, t.Sentence, t.Paragraph,
 		t.Currency, t.Amount, t.AmountWithCurrency,
-		t.ID, t.HyphenatedID,
+		t.HyphenatedID, t.ID,
 	)
 }
 

--- a/faker_test.go
+++ b/faker_test.go
@@ -49,6 +49,8 @@ type SomeStruct struct {
 	Currency           string  `faker:"currency"`
 	Amount             float64 `faker:"amount"`
 	AmountWithCurrency string  `faker:"amount_with_currency"`
+	ID                 string  `faker:"uuid_digit"`
+	HyphenatedID       string  `faker:"uuid_hyphenated"`
 
 	MapStringString        map[string]string
 	MapStringStruct        map[string]AStruct
@@ -107,6 +109,8 @@ type TaggedStruct struct {
 	Currency           string  `faker:"currency"`
 	Amount             float32 `faker:"amount"`
 	AmountWithCurrency string  `faker:"amount_with_currency"`
+	ID                 string  `faker:"uuid_digit"`
+	HyphenatedID       string  `faker:"uuid_hyphenated"`
 }
 
 func (t TaggedStruct) String() string {
@@ -149,6 +153,8 @@ func (t TaggedStruct) String() string {
 	Currency: %s,
 	Amount: %f,
 	AmountWithCurrency: %s,
+	HyphenatedID: %s,
+	ID: %s,
 }`, t.Latitude, t.Longitude, t.CreditCardNumber,
 		t.CreditCardType, t.Email, t.IPV4,
 		t.IPV6, t.Password, t.PhoneNumber, t.MacAddress,
@@ -160,6 +166,7 @@ func (t TaggedStruct) String() string {
 		t.DayOfMonth, t.Timestamp, t.Century, t.TimeZone,
 		t.TimePeriod, t.Word, t.Sentence, t.Paragraph,
 		t.Currency, t.Amount, t.AmountWithCurrency,
+		t.ID, t.HyphenatedID,
 	)
 }
 

--- a/uuid.go
+++ b/uuid.go
@@ -19,11 +19,6 @@ func GetIdentifier() Identifier {
 	return identifier
 }
 
-// SetIdentifier sets
-func SetIdentifier(id Identifier) {
-	identifier = id
-}
-
 // Identifier ...
 type Identifier interface {
 	Digit() string
@@ -34,35 +29,26 @@ type Identifier interface {
 type UUID struct{}
 
 // createUUID returns a 16 byte slice with random values
-func createUUID() ([]byte, error) {
+func createUUID() []byte {
 	b := make([]byte, 16)
-	_, err := io.ReadFull(rand.Reader, b)
-	if err != nil {
-		return nil, err
-	}
+	io.ReadFull(rand.Reader, b)
 	// variant bits; see section 4.1.1
 	b[8] = b[8]&^0xc0 | 0x80
 	// version 4 (pseudo-random); see section 4.1.3
 	b[6] = b[6]&^0xf0 | 0x40
-	return b, nil
+	return b
 }
 
 // Hyphenated returns a 36 byte hyphenated UUID
 func (i UUID) Hyphenated() string {
-	b, err := createUUID()
-	if err != nil {
-		panic(err)
-	}
+	b := createUUID()
 	uuid := fmt.Sprintf("%x-%x-%x-%x-%x", b[0:4], b[4:6], b[6:8], b[8:10], b[10:])
 	return uuid
 }
 
 // Digit returns a 32 bytes UUID
 func (i UUID) Digit() string {
-	b, err := createUUID()
-	if err != nil {
-		panic(err)
-	}
+	b := createUUID()
 	uuid := fmt.Sprintf("%x", b)
 	return uuid
 }

--- a/uuid.go
+++ b/uuid.go
@@ -7,22 +7,22 @@ import (
 	"reflect"
 )
 
-var uuid Identifier
+var identifier Identifier
 
-// GetUUID returns a new Uuid interface of Uuid
-func GetUUID() Identifier {
+// GetIdentifier returns a new Identifier
+func GetIdentifier() Identifier {
 	mu.Lock()
 	defer mu.Unlock()
 
-	if uuid == nil {
-		uuid = &UUID{}
+	if identifier == nil {
+		identifier = &UUID{}
 	}
-	return uuid
+	return identifier
 }
 
-// SetUUID sets custom UUID
-func SetUUID(identifier UUID) {
-	uuid = identifier
+// SetIdentifier sets
+func SetIdentifier(id Identifier) {
+	identifier = id
 }
 
 // Identifier ...
@@ -34,6 +34,7 @@ type Identifier interface {
 // UUID struct
 type UUID struct{}
 
+// createUUID returns a 16 byte slice with random values
 func createUUID() ([]byte, error) {
 	b := make([]byte, 16)
 	_, err := io.ReadFull(rand.Reader, b)
@@ -47,7 +48,7 @@ func createUUID() ([]byte, error) {
 	return b, nil
 }
 
-// Hyphenated returns a 36 bytes UUID with hyphens
+// Hyphenated returns a 36 byte hyphenated UUID
 func (i UUID) Hyphenated(v reflect.Value) error {
 	b, err := createUUID()
 	if err != nil {
@@ -58,7 +59,7 @@ func (i UUID) Hyphenated(v reflect.Value) error {
 	return nil
 }
 
-// Digit returns a 32 bytes UUID with chars
+// Digit returns a 32 bytes UUID
 func (i UUID) Digit(v reflect.Value) error {
 	b, err := createUUID()
 	if err != nil {

--- a/uuid.go
+++ b/uuid.go
@@ -4,7 +4,6 @@ import (
 	"crypto/rand"
 	"fmt"
 	"io"
-	"reflect"
 )
 
 var identifier Identifier
@@ -27,8 +26,8 @@ func SetIdentifier(id Identifier) {
 
 // Identifier ...
 type Identifier interface {
-	Digit(v reflect.Value) error
-	Hyphenated(v reflect.Value) error
+	Digit() string
+	Hyphenated() string
 }
 
 // UUID struct
@@ -49,23 +48,21 @@ func createUUID() ([]byte, error) {
 }
 
 // Hyphenated returns a 36 byte hyphenated UUID
-func (i UUID) Hyphenated(v reflect.Value) error {
+func (i UUID) Hyphenated() string {
 	b, err := createUUID()
 	if err != nil {
-		return err
+		panic(err)
 	}
 	uuid := fmt.Sprintf("%x-%x-%x-%x-%x", b[0:4], b[4:6], b[6:8], b[8:10], b[10:])
-	v.Set(reflect.ValueOf(uuid))
-	return nil
+	return uuid
 }
 
 // Digit returns a 32 bytes UUID
-func (i UUID) Digit(v reflect.Value) error {
+func (i UUID) Digit() string {
 	b, err := createUUID()
 	if err != nil {
-		return err
+		panic(err)
 	}
 	uuid := fmt.Sprintf("%x", b)
-	v.Set(reflect.ValueOf(uuid))
-	return nil
+	return uuid
 }

--- a/uuid.go
+++ b/uuid.go
@@ -1,0 +1,70 @@
+package faker
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"reflect"
+)
+
+var uuid Identifier
+
+// GetUUID returns a new Uuid interface of Uuid
+func GetUUID() Identifier {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if uuid == nil {
+		uuid = &UUID{}
+	}
+	return uuid
+}
+
+// SetUUID sets custom UUID
+func SetUUID(identifier UUID) {
+	uuid = identifier
+}
+
+// Identifier ...
+type Identifier interface {
+	Digit(v reflect.Value) error
+	Hyphenated(v reflect.Value) error
+}
+
+// UUID struct
+type UUID struct{}
+
+func createUUID() ([]byte, error) {
+	b := make([]byte, 16)
+	_, err := io.ReadFull(rand.Reader, b)
+	if err != nil {
+		return nil, err
+	}
+	// variant bits; see section 4.1.1
+	b[8] = b[8]&^0xc0 | 0x80
+	// version 4 (pseudo-random); see section 4.1.3
+	b[6] = b[6]&^0xf0 | 0x40
+	return b, nil
+}
+
+// Hyphenated returns a 36 bytes UUID with hyphens
+func (i UUID) Hyphenated(v reflect.Value) error {
+	b, err := createUUID()
+	if err != nil {
+		return err
+	}
+	uuid := fmt.Sprintf("%x-%x-%x-%x-%x", b[0:4], b[4:6], b[6:8], b[8:10], b[10:])
+	v.Set(reflect.ValueOf(uuid))
+	return nil
+}
+
+// Digit returns a 32 bytes UUID with chars
+func (i UUID) Digit(v reflect.Value) error {
+	b, err := createUUID()
+	if err != nil {
+		return err
+	}
+	uuid := fmt.Sprintf("%x", b)
+	v.Set(reflect.ValueOf(uuid))
+	return nil
+}

--- a/uuid_test.go
+++ b/uuid_test.go
@@ -24,3 +24,10 @@ func TestHyphenated(t *testing.T) {
 	}
 
 }
+
+func TestGetIdentifier(t *testing.T) {
+	identifier := GetIdentifier()
+	if identifier == nil {
+		t.Fatalf("TestGetIdentifier failed because identifier was nil")
+	}
+}

--- a/uuid_test.go
+++ b/uuid_test.go
@@ -21,7 +21,7 @@ type MockUUID struct {
 }*/
 
 func TestDigit(t *testing.T) {
-	p := GetUUID()
+	p := GetIdentifier()
 	mock := MockUUID{}
 	val := reflect.ValueOf(&mock.UUID)
 	err := p.Digit(val.Elem())
@@ -34,7 +34,7 @@ func TestDigit(t *testing.T) {
 }
 
 func TestHyphenated(t *testing.T) {
-	p := GetUUID()
+	p := GetIdentifier()
 	mock := MockUUID{}
 	val := reflect.ValueOf(&mock.UUIDHyphenated)
 	err := p.Hyphenated(val.Elem())

--- a/uuid_test.go
+++ b/uuid_test.go
@@ -1,0 +1,50 @@
+package faker
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"testing"
+)
+
+const (
+	Tag = `faker:"uuid_digit"`
+)
+
+type MockUUID struct {
+	UUID           string `faker:"uuid_digit"`
+	UUIDHyphenated string `faker:"uuid_hyphenated"`
+}
+
+/*func TestSetUUID(t *testing.T) {
+	SetUUID(NogetAndet{})
+}*/
+
+func TestDigit(t *testing.T) {
+	p := GetUUID()
+	mock := MockUUID{}
+	val := reflect.ValueOf(&mock.UUID)
+	err := p.Digit(val.Elem())
+	if err != nil {
+		t.Error(err)
+	}
+	if match, err := regexp.Match("^[a-zA-Z0-9]{32}$", []byte(val.Elem().String())); !match || err != nil {
+		t.Errorf("Could not match the UUID format, err: %+v, match: %+v", err, match)
+	}
+}
+
+func TestHyphenated(t *testing.T) {
+	p := GetUUID()
+	mock := MockUUID{}
+	val := reflect.ValueOf(&mock.UUIDHyphenated)
+	err := p.Hyphenated(val.Elem())
+	if err != nil {
+		t.Error(err)
+	}
+	exp := "[a-zA-Z 0-9]"
+	pattern := fmt.Sprintf("^%s{8}-%s{4}-%s{4}-%s{4}-%s{12}$", exp, exp, exp, exp, exp)
+	if match, err := regexp.Match(pattern, []byte(val.Elem().String())); !match || err != nil {
+		t.Errorf("Could not match the UUID hyphenated format, err: %+v, match: %+v", err, match)
+	}
+
+}

--- a/uuid_test.go
+++ b/uuid_test.go
@@ -2,48 +2,24 @@ package faker
 
 import (
 	"fmt"
-	"reflect"
 	"regexp"
 	"testing"
 )
 
-const (
-	Tag = `faker:"uuid_digit"`
-)
-
-type MockUUID struct {
-	UUID           string `faker:"uuid_digit"`
-	UUIDHyphenated string `faker:"uuid_hyphenated"`
-}
-
-/*func TestSetUUID(t *testing.T) {
-	SetUUID(NogetAndet{})
-}*/
-
 func TestDigit(t *testing.T) {
 	p := GetIdentifier()
-	mock := MockUUID{}
-	val := reflect.ValueOf(&mock.UUID)
-	err := p.Digit(val.Elem())
-	if err != nil {
-		t.Error(err)
-	}
-	if match, err := regexp.Match("^[a-zA-Z0-9]{32}$", []byte(val.Elem().String())); !match || err != nil {
+	uuid := p.Digit()
+	if match, err := regexp.Match("^[a-zA-Z0-9]{32}$", []byte(uuid)); !match || err != nil {
 		t.Errorf("Could not match the UUID format, err: %+v, match: %+v", err, match)
 	}
 }
 
 func TestHyphenated(t *testing.T) {
 	p := GetIdentifier()
-	mock := MockUUID{}
-	val := reflect.ValueOf(&mock.UUIDHyphenated)
-	err := p.Hyphenated(val.Elem())
-	if err != nil {
-		t.Error(err)
-	}
+	uuid := p.Hyphenated()
 	exp := "[a-zA-Z 0-9]"
 	pattern := fmt.Sprintf("^%s{8}-%s{4}-%s{4}-%s{4}-%s{12}$", exp, exp, exp, exp, exp)
-	if match, err := regexp.Match(pattern, []byte(val.Elem().String())); !match || err != nil {
+	if match, err := regexp.Match(pattern, []byte(uuid)); !match || err != nil {
 		t.Errorf("Could not match the UUID hyphenated format, err: %+v, match: %+v", err, match)
 	}
 

